src/main/java/androidx/widget/TextView.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.widget
+
+import android.graphics.drawable.Drawable
+import android.support.annotation.DrawableRes
+import android.support.annotation.RequiresApi
+import android.support.v4.content.ContextCompat
+import android.widget.TextView
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables.
+ *
+ * @see TextView.setCompoundDrawables
+ */
+fun TextView.updateCompoundDrawables(
+    start: Drawable? = compoundDrawables[0],
+    top: Drawable? = compoundDrawables[1],
+    end: Drawable? = compoundDrawables[2],
+    bottom: Drawable? = compoundDrawables[3]
+) {
+    setCompoundDrawables(start, top, end, bottom)
+}
+
+/**
+ * Updates this TextView's Drawables. This version of the method allows using named parameters
+ * to just set one or more Drawables. Use 0 if you do not want a Drawable there.
```

Comment:
Note that this is not consistent with framework's behavior. In particular, passing 0 to a variant of `setCompoundDrawables` will actually throw a `android.content.res.Resources$NotFoundException: Resource ID #0x0` exception.

The question is: Is it OK for android-ktx to enhance/tweak the framework's behavior? In particular, is it OK for `TextView#setCompoundDrawablesWithIntrinsicBounds(0, 0, 0, 0)` to throw an `Exception` but `TextView#updateCompoundDrawablesWithIntrinsicBounds(0, 0, 0, 0)` to execute and clear the compound Drawables?

Reply:
Note that I don't think there's a way to work around this 😢

Reply:
I think this is where we can safely diverge from the base framework's behavior. We should expand on this in the documentation though to explain exactly what happens when you pass 0.

Reply:
Thank you for your insight @cyrilmottier . This divergence was a conscious decision of mine up for review by the owners of this library if a better solution can be applied. I am open to suggestions.

With regards to the documentation of this method, does it not explain it clearly enough? `Use 0 if you do not want a Drawable there.` . I am not sure what more could we add here @romainguy ?

Reply:
Maybe also mention that -1 will be no-op?

Reply:
` Use 0 if you do not want a Drawable there; -1 keeps existing.` ?

Reply:
Maybe something like:

Two special values can be used instead of regular resource IDs:
- 0 to remove an existing drawable (ignored if no drawable exists)
- -1 to keep an existing drawable

Maybe define two constants to make it more readable?